### ISSUE TITLE
[FEATURE REQUEST] Implement enhanced training hparams

### ISSUE THREAD

**Issue by KristinaUlicna** [alan-turing-institute/grace#146]
Enhance the training options by allowing more config hparams:

+ learning rate scheduler - type:
    + StepLR
    + ExpoLR
+ scheduler step
+ scheduler gamma
+ weight decay
